### PR TITLE
improve(gasPriceOracle): Updates & add Linea

### PR DESCRIPTION
**Pull request by pxrl**
- Minor refactorings & simplifications.
- Tighter scoping for adapter exports.
- Remove unnecessary async/await statements.
- Add fallback strategy for OP stack implementations that are not manually defined. This eases new-chain integrations.

While here, establish a placeholder for a custom Linea gas pricing strategy, to be supported once it is made available on mainnet.

### CODE REVIEW

File: src/gasPriceOracle/oracle.ts [hunk 1]
```diff
@@ -14,29 +19,29 @@ export async function getGasPriceEstimate(
...
-    10: eip1559,
-    137: polygonGasStation,
-    288: legacy,
-    324: legacy,
-    8453: eip1559,
-    42161: eip1559_arbitrum,
+    [CHAIN_IDs.ARBITRUM]: arbitrum.eip1559,
+    [CHAIN_IDs.BASE]: ethereum.eip1559,
+    [CHAIN_IDs.BOBA]: ethereum.legacy,
+    [CHAIN_IDs.LINEA]: linea.eip1559, // @todo: Support linea_estimateGas in adapter.
```
Comment (pxrl) [inline src/gasPriceOracle/oracle.ts:30]: I've added ACX-2269 to remind us to do this once it's available on Linea mainnet.

File: src/gasPriceOracle/oracle.ts [hunk 2]
```diff
@@ -14,29 +19,29 @@ export async function getGasPriceEstimate(
...
   };
 
-  let gasPriceFeed: GasPriceFeed = gasPriceFeeds[chainId];
+  let gasPriceFeed = gasPriceFeeds[chainId];
   if (gasPriceFeed === undefined) {
     if (!legacyFallback) {
       throw new Error(`No suitable gas price oracle for Chain ID ${chainId}`);
     }
-    gasPriceFeed = legacy;
+    gasPriceFeed = chainIsOPStack(chainId) ? ethereum.eip1559 : ethereum.legacy;
```
Comment (pxrl) [inline src/gasPriceOracle/oracle.ts:43]: We'll need to amend this later when other stacks are added, but for now it'll work fine for OP stack.

**Review by bmzig — APPROVED** [review]
Makes sense.

**Review by dohaki — APPROVED** [review]
LGTM

**Pull request merged**